Support multiarch installation
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 2.8.5)
 
 project(powercap)
 set(VERSION_MAJOR 0)
@@ -8,6 +8,8 @@
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -std=c99 -D_GNU_SOURCE")
 
+include(GNUInstallDirs)
+
 # See powercap-common.h for enumeration
 set(POWERCAP_LOG_LEVEL 2 CACHE STRING "Set the log level: 0=DEBUG, 1=INFO, 2=WARN (default), 3=ERROR, 4=OFF")
 
@@ -52,8 +54,8 @@
 # pkg-config
 
 set(PKG_CONFIG_EXEC_PREFIX "\${prefix}")
-set(PKG_CONFIG_LIBDIR "\${prefix}/lib")
-set(PKG_CONFIG_INCLUDEDIR "\${prefix}/include/${PROJECT_NAME}")
+set(PKG_CONFIG_LIBDIR "\${prefix}/${CMAKE_INSTALL_LIBDIR}")
+set(PKG_CONFIG_INCLUDEDIR "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}")
 set(PKG_CONFIG_CFLAGS "-I\${includedir}")
 
 set(PKG_CONFIG_NAME "${PROJECT_NAME}")
@@ -67,9 +69,9 @@
 
 # Install
 
-install(TARGETS powercap DESTINATION lib)
-install(FILES inc/powercap.h inc/powercap-sysfs.h inc/powercap-rapl.h inc/powercap-rapl-sysfs.h DESTINATION include/${PROJECT_NAME})
-install(DIRECTORY ${CMAKE_BINARY_DIR}/pkgconfig/ DESTINATION lib/pkgconfig)
+install(TARGETS powercap DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES inc/powercap.h inc/powercap-sysfs.h inc/powercap-rapl.h inc/powercap-rapl-sysfs.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
+install(DIRECTORY ${CMAKE_BINARY_DIR}/pkgconfig/ DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
 # Uninstall
 
--- a/utils/CMakeLists.txt
+++ b/utils/CMakeLists.txt
@@ -14,5 +14,5 @@
 
 # Install
 
-install(TARGETS powercap-info powercap-set rapl-info rapl-set DESTINATION bin)
-install(DIRECTORY man/ DESTINATION share/man)
+install(TARGETS powercap-info powercap-set rapl-info rapl-set DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(DIRECTORY man/ DESTINATION ${CMAKE_INSTALL_MANDIR})
